Indexing/TermSharing.cpp

```diff
@@ -45,28 +45,6 @@ TermSharing::TermSharing()
 {
 }
 
-/**
- * Destroy the term sharing structure.
- * @since 29/12/2007 Manchester
- */
-TermSharing::~TermSharing()
-{
-#if CHECK_LEAKS
-  Set<Term*,TermSharing>::Iterator ts(_terms);
-  while (ts.hasNext()) {
-    ts.next()->destroy();
-  }
-  Set<Literal*,TermSharing>::Iterator ls(_literals);
-  while (ls.hasNext()) {
-    ls.next()->destroy();
-  }
-  Set<AtomicSort*,TermSharing>::Iterator ss(_sorts);
-  while (ss.hasNext()) {
-    ss.next()->destroy();
-  }
-#endif
-}
-
```

Comment:
How about we keep CHECK_LEAKS and, e.g., this bit? Otherwise, the valgrinder will need to write it again?

Reply:
Sure!